test/run-tests

```diff
@@ -861,23 +893,89 @@ def main():
+    args.inventory = os.environ.get("TEST_HARNESS_INVENTORY", args.inventory)
+    args.pull_request = os.environ.get("TEST_HARNESS_PULL_REQUEST", args.pull_request)
+    args.use_images = os.environ.get("TEST_HARNESS_USE_IMAGES", args.use_images)
+    args.dry_run = os.environ.get("TEST_HARNESS_DRY_RUN", args.dry_run)
+    args.keep_results = os.environ.get("TEST_HARNESS_KEEP_RESULTS", args.keep_results)
+
     with open(args.config + "/config.json") as configfile:
         config.update(json.load(configfile))
         images = config["images"]
         random.shuffle(images)
         repos = [r.split("/") for r in config.get("repositories", [])]
         random.shuffle(repos)
 
+    # this is the name for github status reporting
+    if "variant" not in config:
+        if config.get("name") == "linux-system-roles-test-staging":
+            config["variant"] = "staging"
+        else:
+            config["variant"] = ""
+    config["variant"] = os.environ.get("TEST_HARNESS_CONFIG_VARIANT", config["variant"])
+
+    env_repos = os.environ.get("TEST_HARNESS_REPOSITORIES")
+    if env_repos:
+        repos = [r.split("/") for r in env_repos.split(",")]
     setup_logging(config.get("logging", {}))
 
     check_environment()
 
+    ansible_version = LooseVersion(get_ansible_version())
+    logging.info("Using Ansible version {}".format(ansible_version))
+    # this is used in PR status - context name
+    ansible_id = f"ansible-{ansible_version.version[0]}.{ansible_version.version[1]}"
+
+    image_patterns = args.use_images.split(",")
+    # copy all images to test_images . . .
+    test_images = dict([(image["name"], image) for image in images])
+    # . . . then remove the ones that do not match the criteria
+    for image in images:
+        min_ansible_version = LooseVersion(image.get("min_ansible_version", "0"))
+        if ansible_version < min_ansible_version:
+            if image["name"] in test_images:
+                logging.debug(
+                    f"Image {image['name']} will not be used for testing because "
+                    f"the minimum Ansible version {min_ansible_version} required "
+                    f"by the image is greater than the version {ansible_version} "
+                    "of Ansible used by the test."
+                )
+                del test_images[image["name"]]
+            continue
+        for pattern in image_patterns:
+            if not (
+                fnmatch.fnmatch(image["name"], pattern)
+                or fnmatch.fnmatch(image["source"], pattern)
+            ):
+                if image["name"] in test_images:
+                    logging.debug(
+                        f"Image {image['name']} will not be used for testing "
+                        f"because neither the name nor the source match the "
+                        f"image pattern {pattern}"
+                    )
+                    del test_images[image["name"]]
+                continue
```

Comment:
This does not require that all patterns match for an image to be used because as soon as one image does not match, the image will be removed from the test_images, for example patterns `a,b` will remove both the images `a` and `b` IIUC.